docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Elytron_OIDC_Client.adoc

```diff
@@ -173,6 +173,195 @@ It is also possible to configure the scope values that would be used via the sco
+"secp256r1" if you are planning on using the "ES256" algorithm, "secp384r1" for "ES384" and so on. However, for RSA keys,
+keys of size 2048 would be applicable for the supported RSXXX and PSXXX algorithms regardless of the SHA value.
+
+==== Configuring Client Keys for Keycloak
+
+In order for the OpenID provider to verify the signature on the JWT, the public key needs to be communicated with the
+OpenID provider. In keycloak, this can be done by switching the `public-client` property to false when configuring the
+`elytron-oidc-client` subsystem (or on the deployment configuration), switching `client-configuration` to `on` on the
+keycloak console and importing the certificate from the keystore under the `keys` tab. You can also upload the certificate
+using a JWKSUrl on the keycloak console. However, when specifying the keystore for setting up the server, WildFly does
+not support the use of a JWKSUrl.
+
+In order to configure client keys on the Keycloak console, the `Client configuration` toggle needs to be turned on. This
+will add a new tab for the client called `Keys`, where we can import the client keys. Please note that the keycloak
+client will store the certificate using inside `jwt.credential.certificate` attribute and it only stores one certificate
+at a time, even if the JWKSUrl has more than one key.
+
+Toggling `Client configuration` to true turns the `Public-client` attribute to `false`. Therefore, when setting up the
+WildFly server, the `public-client` attribute must also be set to `false` and client credentials (i.e. client secret)
+will need to be configured.
+
+==== Encrypting the Request object
+
+Request Objects can be optionally encrypted using one of the encryption and content encryption algorithms supported
+by the OpenID provider. In order to specify these algorithms when configuring the server, the
+`request-object-encryption-alg-value` and the `request-object-encryption-enc-value` attributes must be
+specified for the JWT to be encrypted. If only one is specified, then both will remain undefined and the JWT will not be
+encrypted. These two attributes are of type string and their default value will be undefined.
+
+No additional keys need to be configured for this, since the JWT will be encrypted using the public realm key that the
+OpenID provider will share. The JWT will be signed first and then encrypted as a nested JWT. It is possible to mix and
+match between the types of algorithms for signing and encrypting. For example, a JWT can be signed using elliptical
+algorithms and encrypted using RSA algorithms.
+
+NOTE: Not all OpenID providers will support encrypted request objects. Please refer to the OpenID metadata, specifically
+`request_object_encryption_alg_values_supported` and `request_object_encryption_enc-values_supported` sections for
+supported algorithms. If this section is not present, the OpenID provider does not support encrypted request objects.
+
+==== Server Configuration for "Request" Parameter
```

Comment:
Is this section meant to show example configuration? Maybe we could rename it.

Reply:
It mentions server config but shows both subsystem and deployment config so maybe example configuration is a better name.

Reply:
Or maybe this could be part of the use-case driven guide?

Reply:
Thanks @fjuma 
I have addressed the comments above. I also ran the full wildfly testsuite and it passed!